e2e/openshift/openshift_packaging_test.go

```diff
@@ -0,0 +1,340 @@
+// This is the YAML used to create an operatorGroup
+// in OpenShift to deploy the operator.
+const operatorGroup = `apiVersion: operators.coreos.com/v1
+kind: OperatorGroup
+metadata:
+  name: test
+  namespace: test
+spec:
+  targetNamespaces:
+  - test
+`
+
+// This is the YAML used to create a CatalogSource
+// in OpenShift to have the operator appear in the Marketplace
+const catalogSource = `apiVersion: operators.coreos.com/v1alpha1
+kind: CatalogSource
+metadata:
+  name: cockroach-test
+  namespace: openshift-marketplace
+spec:
+  sourceType: grpc
+  displayName: Cockroach Test Demo
+  image: {{.DockerRegistry}}/cockroachdb-operator-index:{{.AppVersion}}`
+
+// This is the YAML used to create a subscription
+// in OpenShift to deploy the operator.
+const subscription = `apiVersion: operators.coreos.com/v1alpha1
+kind: Subscription
+metadata:
+  name: myoperator
+  namespace: test
+spec:
+  channel: stable
+  installPlanApproval: Automatic
+  name: cockroachdb-certified
+  source: cockroach-test
+  sourceNamespace: openshift-marketplace
+  startingCSV: cockroach-operator.{{.AppVersion}}`
+
+// YAMLTemplate struct is used to render the
+// above YAML files
+type YAMLTemplate struct {
+	AppVersion     string
+	DockerRegistry string
+}
+
+// delete is a helper variable used to delete resources
+var delete = []string{
+	"delete",
+	"-n",
+}
+
+// TestPackage is used to install the operator in an OpenShift
+// cluster. It uses three different YAML files and uses the test
+// namespace. This program uses oc and also installs example.yaml
+// to test that the operator is functioning.
+func TestPackaging(t *testing.T) {
+
+	// Test that three env variables are set
+	appVersion := os.Getenv("APP_VERSION")
+	require.True(t, appVersion != "", "APP_VERSION env var not set")
+	dockerRegistry := os.Getenv("DOCKER_REGISTRY")
+	require.True(t, dockerRegistry != "", "DOCKER_REGISTRY env var not set")
+	kubeconfig := os.Getenv("KUBECONFIG")
+	require.True(t, kubeconfig != "", "KUBECONFIG evn var not set")
+
+	// remove old cluster if it still exists
+	args := []string{
+		"delete",
+		"crdbclusters.crdb.cockroachlabs.com",
+		"cockroachdb",
+	}
+
+	process.ExecJUnit("oc", args, os.Environ())
+
+	// remove old pvc's from last run
+	args = []string{
+		"-n", "test",
+		"delete",
+		"pvc",
+		"-l",
+		"app.kubernetes.io/name=cockroachdb",
+	}
+
+	process.ExecJUnit("oc", args, os.Environ())
+
+	// remove test namespace
+	args = []string{
+		"delete",
+		"namespace",
+		"test",
+	}
+
+	process.ExecJUnit("oc", args, os.Environ())
+
+	// remove crds
+	args = []string{
+		"delete",
+		"crd",
+		"crdbclusters.crdb.cockroachlabs.com",
+	}
+
+	process.ExecJUnit("oc", args, os.Environ())
+
+	// create a new project/namespace for testing
+	args = []string{
+		"new-project",
+		"test",
+	}
+
+	// we cannot do a require.NoError on process, because oc tries to modify
+	// the kubeconfig file, and with bazel you cannot do this
+	process.ExecJUnit("oc", args, os.Environ())
```

Comment:
So I understand, the error thrown here will be bazel complaining that `kubeconfig` was modified? Is this the case for the other junit execs?

Reply:
oc tries to change the kubeconfig file and bazel does not allow it to, since the file is outside of the sandbox. So oc throws an error and the test fails. The project is created successfully. So the oc command does what we needed it to do. We could try to create a namespace rather than creating a new project, but I do not know if that would work. Oh we could try to copy the kubeconfig into the sandbox.

The next steps in the test will fail if the ns is missing.